libunwind/src/UnwindRegistersRestore.S

```diff
@@ -731,8 +712,18 @@ WEAK_ALIAS(__rtld_unw_setcontext, _rtld_unw_setcontext)
   .p2align 2
 DEFINE_LIBUNWIND_FUNCTION(__libunwind_Registers_arm64_jumpto)
 #ifdef __CHERI_PURE_CAPABILITY__
+#ifdef _LIBUNWIND_HAS_CHERI_LIB_C18N
+  // Preserve the argument in a callee-saved register instead of pushing it onto
+  // the stack because stack unwinding will switch the stack.
```

Comment:
Whilst using a saved register is better anyway, this justification seems undesirable, shouldn't it leave us on the same stack? Then when this function / longjmp returns it can restore the real stack that corresponds to what the compartment should have.

Reply:
We can leave the stack unchanged, but then there would be a small window between after `dl_c18n_unwind_trusted_stk` returns and the new stack is installed where the trusted stack says we're in the target compartment but the stack pointer still contains that of the source compartment. This doesn't seem desirable.

Under the current implementation, after `dl_c18n_unwind_trusted_stk` returns, we are for all purposes already in the target compartment (i.e. not even in libunwind's compartment anymore). And the final `ret` resumes execution of the target compartment without going through a trampoline. (Because `unw_getcontext` is a trusted function, it has access to the raw return capability to the target compartment.)